app/src/main/java/org/p2p/wallet/user/repository/UserInMemoryRepository.kt

```diff
@@ -94,14 +94,12 @@ class UserInMemoryRepository(
 
         // Filter items that start with the query
         allTokens.filterTo(filteredList) {
-            it.symbol.startsWith(searchText, ignoreCase = true) || it.name.startsWith(searchText, ignoreCase = true)
+            val isNamesEqualToSearchText = it.name.equals(searchText, ignoreCase = true)
+            val isNameStartsWithSearchText = it.name.startsWith(searchText, ignoreCase = true)
+            val isSymbolStartsWithSearchText = it.symbol.startsWith(searchText, ignoreCase = true)
+            isNamesEqualToSearchText || isNameStartsWithSearchText || isSymbolStartsWithSearchText
         }
 
-        // Filter items that contain the query
-        allTokens.filterTo(filteredList) {
-            it.symbol.contains(searchText, ignoreCase = true) && !it.symbol.startsWith(searchText, ignoreCase = true) ||
-                it.name.contains(searchText, ignoreCase = true) && !it.name.startsWith(searchText, ignoreCase = true)
-        }
```

Comment:
We need the items which start with the query to be the first in the list. 
And the items which contains the query after